vm-virtio/src/vhost_user/handler.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Portions Copyright 2017 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the THIRD-PARTY file.
+
+use super::super::{Queue, VirtioInterruptType};
+use super::Error;
+use epoll;
+use vmm_sys_util::eventfd::EventFd;
+
+use crate::VirtioInterrupt;
+use std::io::{self};
+use std::os::unix::io::AsRawFd;
+use std::result;
+use std::sync::Arc;
+
+pub type Result<T> = result::Result<T, Error>;
+
+pub struct VhostUserEpollHandler {
+    pub interrupt_cb: Arc<VirtioInterrupt>,
+    pub kill_evt: EventFd,
+    pub queues: Vec<Queue>,
+    pub vu_interrupt_list: Vec<EventFd>,
+}
+
+impl VhostUserEpollHandler {
```

Comment:
I'm not opposed to have a dedicated `VhostUserEpollHandler`, but please make sure it can be used from virtio-fs (`fs.rs`). I'd like to make sure we come up with some clean factorization instead of duplicating with existing code.